shell/platform/tizen/tizen_input_method_context.cc

```diff
@@ -83,19 +83,33 @@ Ecore_IMF_Keyboard_Locks EcoreInputModifiersToEcoreImfLocks(
 }
 
 template <typename T>
-T EcoreEventKeyToEcoreImfEvent(Ecore_Event_Key* event, const char* dev_name) {
+T EcoreEventKeyToEcoreImfEvent(Ecore_Event_Key* event) {
   T imf_event;
 
   imf_event.keyname = event->keyname;
   imf_event.key = event->key;
   imf_event.string = event->string;
   imf_event.compose = event->compose;
   imf_event.timestamp = event->timestamp;
+  imf_event.keycode = event->keycode;
+
   imf_event.modifiers =
       EcoreInputModifiersToEcoreImfModifiers(event->modifiers);
   imf_event.locks = EcoreInputModifiersToEcoreImfLocks(event->modifiers);
-  imf_event.dev_name = dev_name;
-  imf_event.keycode = event->keycode;
+
+  if (event->dev) {
+    imf_event.dev_name = ecore_device_name_get(event->dev)
+                             ? ecore_device_name_get(event->dev)
```

Comment:
How about using a local variable rather than calling the same function twice?
like this?
```
const char* device_name = ecore_device_name_get(event->dev);
imf_event.dev_name = device_name ? device_name : "";
```

Reply:
I updated it.